### PR TITLE
FEAT: add shared to cmake extension

### PR DESCRIPTION
**Pull request by MementoRC**
This adds a verification flow to check that a shared build works

### CODE REVIEW

**Comment by ofek** [comment]
Is this still relevant? I'm sorry I don't have much time to look into it! If so I will merge.

**Comment by MementoRC** [comment]
Well, yes it is. It only adds a n additional workflow to verify that shared library can be used. It is not urgent or critical

File: .github/workflows/verify_shared_build.yml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      with:
+        python-version: '3.12'
+
+    - name: Build wheels
+      uses: pypa/cibuildwheel@v2.17
+      env:
+        COINCURVE_CROSS_HOST: 'x86'
+        CIBW_ARCHS_WINDOWS: 'x86'
+        CIBW_BEFORE_ALL: choco install -y --no-progress --no-color cmake>=3.28 pkgconfiglite
+
```
Comment (ofek) [inline .github/workflows/verify_shared_build.yml:157]:
```suggestion
        CIBW_BEFORE_ALL: choco install -y --no-progress --no-color cmake>=3.28
```

File: .github/workflows/verify_shared_build.yml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Build wheels
+      uses: pypa/cibuildwheel@v2.17
+      env:
+        COINCURVE_CROSS_HOST: 'arm64'
+        CIBW_ARCHS_WINDOWS: 'ARM64'
+        CIBW_BEFORE_ALL: choco install -y --no-progress --no-color cmake>=3.28 pkgconfiglite
```
Comment (ofek) [inline .github/workflows/verify_shared_build.yml:151]:
```suggestion
        CIBW_BEFORE_ALL: choco install -y --no-progress --no-color cmake>=3.28
```

**Review by ofek — APPROVED** [review]

**Comment by ofek** [comment]
What's this, a hash of the source tar ball? https://github.com/ofek/coincurve/blob/adff792cc1fe0cfa634fdd7c584dedee17107874/pyproject.toml#L82

**Comment by MementoRC** [comment]
Yes, I think I used it in the CMake fetching the secp256k1 tarball. It's likely the `sha256sum`

**Pull request merged**